src/Carousel.php

```diff
@@ -20,244 +22,535 @@
      *
-     * @param array|string $item a single item from {@see items}
-     * @param int $index the item index as the first item should be set to `active`
+     * @return string The HTML representation of the element.
+     */
+    private function renderControlPrev(string $id): string
+    {
+        return Button::button('')
+            ->addAttributes(
+                [
+                    'data-bs-target' => '#' . $id,
+                    'data-bs-slide' => 'prev',
+                ],
+            )
+            ->addClass(self::CLASS_CAROUSEL_CONTROL_PREV)
+            ->addContent(
+                "\n",
+                Span::tag()
+                    ->addAttributes(['aria-hidden' => 'true'])
+                    ->addClass(self::CLASS_CAROUSEL_CONTROL_PREV_ICON)
+                    ->render(),
+                "\n",
+                Span::tag()->addClass('visually-hidden')->addContent($this->controlPrevLabel)->render(),
+                "\n",
+            )
+            ->encode(false)
+            ->render();
+    }
+
+    /**
+     * Renders the carousel indicators.
      *
-     * @throws JsonException|RuntimeException if the item is invalid.
+     * @param string $id The ID of the carousel component.
      *
-     * @return string the rendering result.
+     * @return string The HTML representation of the element.
      */
-    private function renderItem(array|string $item, int $index): string
+    private function renderIndicator(int $key, CarouselItem $carouselItem, string $id): string
     {
-        if (is_string($item)) {
-            $content = $item;
-            $caption = null;
-            $options = [];
-        } elseif (isset($item['content'])) {
-            $content = $item['content'];
-            $caption = ArrayHelper::getValue($item, 'caption');
-
-            if ($caption !== null) {
-                $captionOptions = ArrayHelper::remove($item, 'captionOptions', []);
-                Html::addCssClass($captionOptions, ['captionOptions' => 'carousel-caption']);
-
-                $caption = Html::div($caption, $captionOptions)
-                    ->encode($this->encodeTags)
-                    ->render();
+        return Button::button('')
+            ->addAttributes(
+                [
+                    'data-bs-target' => '#' . $id,
+                    'data-bs-slide-to' => (string) $key,
+                    'aria-current' => $carouselItem->isActive() ? 'true' : null,
+                    'aria-label' => 'Slide ' . (string) ($key + 1),
+                ],
+            )
+            ->addClass($carouselItem->isActive() ? 'active' : null)
+            ->encode(false)
+            ->render();
+    }
+
+    /**
+     * Renders the carousel items.
+     *
+     * @param string $id The ID of the carousel component.
+     *
+     * @return string The HTML representation of the element.
+     */
+    private function renderItems(string $id): string
```

Comment:
If no active items, it needs use first item as active.

Reply:
done.